web/docs/auth/entities/entities.md

```diff
@@ -1,22 +1,454 @@
+            include: {
+              identities: {
+                // Making sure to exclude the `providerData` field
+                select: {
+                  providerName: true,
+                  providerUserId: true,
+                },
+              },
+            },
+          },
+        },
+      },
+      // highlight-end
+    },
+  })
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```ts title="src/tasks.ts"
+export const getAllTasks = (async (args, context) => {
+  return context.entities.Task.findMany({
+    orderBy: { id: 'desc' },
+    select: {
+      id: true,
+      title: true,
+      // highlight-start
+      user: {
+        include: {
+          auth: {
+            include: {
+              identities: {
+                // Making sure to exclude the `providerData` field
+                select: {
+                  providerName: true,
+                  providerUserId: true,
+                },
+              },
+            },
+          },
+        },
+      },
+      // highlight-end
+    },
+  })
+}) satisfies tasks.GetAllQuery<{}, {}>
+```
+
+</TabItem>
+</Tabs>
+
+If you have some **piece of the auth data that you want to access frequently** (for example the `username`), it's best to store it at the top level of the `User` entity.
+
+For example, save the `username` or `email` as a property on the `User` and you'll be able to access it without fetching the auth-related fields. We show an example in the [Defining Extra Fields on the User Entity](../overview.md#1-defining-extra-fields) section of the docs.
+
+### Getting the User's Data
+
+If you fetch the user data like this, you'll lose the benefits of the `AuthUser` object. You'll need to access the auth data by going through the `auth` and `identities` relations:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```jsx title="src/MainPage.jsx"
+function MainPage() {
+  // ...
+  return (
+    <div className="tasks">
+      {tasks.map((task) => (
+        <div key={task.id} className="task">
+          {task.title} by {task.user.auth?.identities[0].providerUserId}
+        </div>
+      ))}
+    </div>
+  )
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```tsx title="src/MainPage.tsx"
+function MainPage() {
+  // ...
+  return (
+    <div className="tasks">
+      {tasks.map((task) => (
+        <div key={task.id} className="task">
+          {task.title} by {task.user.auth?.identities[0].providerUserId}
+        </div>
+      ))}
+    </div>
+  )
+}
+```
+
+</TabItem>
+</Tabs>
+
+Wasp offers a few helper methods to access the user's data when you retrieve the `user` like this. They are `getUsername`, `getEmail` and `getFirstProviderUserId`. They can be used both on the client and the server.
+
+#### `getUsername`
+
+It accepts the `user` object and returns the username of the user if it exists or `null` if it doesn't. The `user` object needs to have the `auth` and the `identities` relations included.
```

Comment:
If I am correct this returns username ONLY as defined by `auth.identities.username`, right? It won't get username from `user.username` if user defined it there on their own, for sure. It also won't get username from their Google signing, or any other auth method.
While this might be clear to us, I don't think this would be clear to the first time reader, so I think we should make this very clear.
Although we do have a separate issue for this here https://github.com/wasp-lang/wasp/issues/1800, so maybe that is ok and we will just fix it there? Same goes for getEmail.